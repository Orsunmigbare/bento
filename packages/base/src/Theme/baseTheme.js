import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
