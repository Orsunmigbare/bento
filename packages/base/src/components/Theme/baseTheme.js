export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
