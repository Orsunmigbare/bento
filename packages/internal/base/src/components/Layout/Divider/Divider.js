import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { color, themify } from "@staccx/theme"
import themePropTypes from "../../constants/themePropTypes"

const Divider = ({ level, className, height, ...restProps }) => {
  return <Rule className={className} height={height} {...restProps} />
}

Divider.themeProps = {
  rule: {
    name: "divider",
    description: "horizontal line style",
    type: themePropTypes.style
  }
}
export const Rule = styled.hr`
  height: ${p => p.height}px;
  width: 100%;
  min-width: 100px;
  border: none;
  background-color: ${color.line};
  ${themify(Divider.themeProps.rule)};
`

Divider.defaultProps = {
  className: null,
  height: 1
}

Divider.propTypes = {
  className: PropTypes.string,
  height: PropTypes.number
}

export default Divider
