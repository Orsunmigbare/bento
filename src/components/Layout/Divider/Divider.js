import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, color, themePropTypes } from "../../../theming"

const Divider = ({ level, className, height, ...restProps }) => {
  return <Rule className={className} height={height} {...restProps} />
}

Divider.themeProps = {
  rule: {
    name: "divider",
    description: "horizontal line style",
    type: themePropTypes.style
  }
}
export const Rule = styled.hr`
  height: ${p => p.height}px;
  width: 100%;
  min-width: 100px;
  border: none;
  background-color: ${color.line};
  ${applyVariants(Divider.themeProps.rule)};
`

Divider.defaultProps = {
  className: null,
  height: 1
}

Divider.propTypes = {
  /**
   * The component can be assigned a class to set the styling.
   */
  className: PropTypes.string,
  /**
   * The height of the Divider.
   */
  height: PropTypes.number
}

export default Divider
