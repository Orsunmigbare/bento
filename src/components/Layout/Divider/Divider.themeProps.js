import { themePropTypes } from "../../../theming"

const themeProps = {
  rule: {
    name: "divider",
    description: "horizontal line style",
    type: themePropTypes.style
  }
}

export default themeProps
