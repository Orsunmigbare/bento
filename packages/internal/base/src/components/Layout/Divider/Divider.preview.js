import React from "react"
import Divider from "./Divider"

const preview = {
  title: "Divider",
  category: "Components/Layout",
  tags: ["layout"],
  component: Divider,
  render: props => <Divider {...props} />
}

export default preview
