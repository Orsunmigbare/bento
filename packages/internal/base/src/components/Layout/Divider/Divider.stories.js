import React from "react"
import { storiesOf } from "@storybook/react"
import { withKnobs, number } from "@storybook/addon-knobs"
import docs from "./Divider.md"
import Divider from "./Divider"

storiesOf("components|base/Layout/Divider", module)
  .addDecorator(withKnobs)
  .add("No props", () => <Divider />, {
    info: {
      text: docs
    }
  })
  .add("Height at 10", () => <Divider height={number("Height", 10)} />, {
    info: {
      text: docs
    }
  })
