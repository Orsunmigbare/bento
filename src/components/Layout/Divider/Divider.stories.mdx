import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks"
import Divider from "./Divider"

<Meta title="Components|Layout/Divider" component={Divider} />

# Divider
This is the Divider component. The component can be used to structure content such as chunks of text, input-groups or to create a thematic section. 


# import

```javascript
import { Divider } from "@staccx/bento"
```

# Types of Dividers
There are several ways to display the Divider component.

## Standard Divider
The standard Divider uses the height of "1", but it can be modified as desired. 

<Preview>
  <Story name="Standard">
    <Divider height="1"/>
  </Story>
</Preview>

## Modified height
The Divider below uses the height of "5".

<Preview>
  <Story name="Modified height">
    <Divider height="5"/>
  </Story>
</Preview>

# Props for Divider

<Props of={Divider} />
