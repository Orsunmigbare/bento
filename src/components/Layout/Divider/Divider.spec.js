import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Divider from "./Divider"
describe("Alert", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Divider />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Height", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Divider height={15} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("className", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Divider className="className" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("All", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Divider className="className" height={10} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
