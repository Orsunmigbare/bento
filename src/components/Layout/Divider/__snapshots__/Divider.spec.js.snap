// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshots All 1`] = `
<hr
  className="sc-AykKC lPswi className"
  height={10}
/>
`;

exports[`Alert Snapshots Height 1`] = `
<hr
  className="sc-AykKC jkDrjb"
  height={15}
/>
`;

exports[`Alert Snapshots No props 1`] = `
<hr
  className="sc-AykKC fyMlMi"
  height={1}
/>
`;

exports[`Alert Snapshots className 1`] = `
<hr
  className="sc-AykKC fyMlMi className"
  height={1}
/>
`;
